### PR TITLE
Drilling long action

### PR DESCRIPTION
**Pull request by Night-Pryanik**
- Changed duration from 500 moves to 30 minutes;
- In `jackhammer` iuse replaced check for `t_tree` to check for more generic `TREE` flag;
- `jackhammer` iuse now uses 3D.

### CODE REVIEW

File: src/iuse.cpp [hunk 1]
```diff
@@ -2945,34 +2945,34 @@ int iuse::jackhammer(player *p, item *it, bool, const tripoint &pos )
...
-        p->moves -= 500;
-        sounds::sound( dirp, 45, _( "TATATATATATATAT!" ) );
+    int turns;
+
+    if( ( g->m.is_bashable( dirp ) &&
+      ( g->m.has_flag( "SUPPORTS_ROOF", dirp ) || g->m.has_flag( "MINEABLE", dirp ) ) &&
+      !g->m.has_flag( "TREE", dirp ) ) ||
+    ( g->m.move_cost( dirp ) == 2 && g->get_levz() != -1 &&
+      g->m.ter( dirp ) != t_dirt && g->m.ter( dirp ) != t_grass ) ) {
+        turns = MINUTES( 30 );
```
Comment (ZhilkinSerg) [inline src/iuse.cpp]: Maybe it is possible to use `30_minutes`?

**Comment by Night-Pryanik** [comment]
We definitely should change `assign_activity` to accept `time duration` values.

**Comment by ecroffleps** [comment]
This fixes #20023, the Issue should probably be closed.

**Comment by Night-Pryanik** [comment]
Yeah, thanks.

**Pull request merged**